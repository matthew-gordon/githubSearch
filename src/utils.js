import React from 'react'
import { StyleSheet } from 'react-native'
import {
  JavaScriptIcon,
  PythonIcon,
  JavaIcon,
  RubyIcon,
  TypeScriptIcon,
} from './components/common/Icons'

export function buildUrl({ baseUrl, resource, searchQuery, languages, sort, order }) {
  let selectedLanguages = ''
  let query = ''

  if (!searchQuery) {
    query = 'github'
  } else {
    query = searchQuery
  }

  if (languages.length > 1) {
    selectedLanguages = languages
      .reduce((prev, curr) => {
        return [...prev, `language:${curr}`]
      }, '')
      .join('+')
  } else {
    selectedLanguages = `language:${languages[0]}`
  }

  return `${baseUrl}/${resource}?q=${query}+${selectedLanguages}&sort=${sort}&order=${order}`
}

export function renderLanguageIcon({ size, language }) {
  switch (language) {
    case 'javascript':
      return <JavaScriptIcon size={size} style={styles.icon} />
    case 'python':
      return <PythonIcon size={size} style={styles.icon} />
    case 'java':
      return <JavaIcon size={size} style={styles.icon} />
    case 'ruby':
      return <RubyIcon size={size} style={styles.icon} />
    case 'typescript':
      return <TypeScriptIcon size={size} style={styles.icon} />
  }
}

const styles = StyleSheet.create({
  icon: {
    margin: 10,
  },
})
