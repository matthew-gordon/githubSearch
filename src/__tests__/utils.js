import { render } from '@testing-library/react-native'
import { renderLanguageIcon, buildUrl } from '../utils'

describe('utils', () => {
  describe('renerLanguageIcon', () => {
    it('renders an icon with correct properties', () => {
      const { baseElement } = render(renderLanguageIcon({ size: 25, language: 'javascript' }))

      expect(baseElement).toMatchSnapshot()
    })
  })

  describe('buildUrl', () => {
    it('builds correct url for single language', () => {
      const url = buildUrl({
        baseUrl: 'https://api.github.com/search',
        resource: 'repositories',
        searchQuery: 'tetris',
        languages: ['javascript'],
        sort: 'stars',
        order: 'desc',
      })

      expect(url).toBe(
        'https://api.github.com/search/repositories?q=tetris+language:javascript&sort=stars&order=desc'
      )
    })
  })
})
