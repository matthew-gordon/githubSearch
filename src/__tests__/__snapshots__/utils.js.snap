// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils renerLanguageIcon renders an icon with correct properties 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 25,
          },
          Object {
            "margin": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ïŒž
    </Text>
  </View>
</View>
`;
